core/templates/paged_allocator.h

```diff
@@ -116,16 +117,41 @@ class PagedAllocator {
 			allocs_available = 0;
 		}
 	}
+
+public:
+	void reset(bool p_allow_unfreed = false) {
+		if (thread_safe) {
+			spin_lock.lock();
+		}
+		_reset(p_allow_unfreed);
+		if (thread_safe) {
+			spin_lock.unlock();
+		}
+	}
```

Comment:
Wonder if it is possible to have a spinlock guard like `MutexLock`?

Reply:
Not a bad idea, but that would be out of the scope of this PR. And, in any case, some of the Godot code is not even using the `MutexLock`, preferring explicit lock-unlock even when the RAII lock would be usable.